### ISSUE TITLE
Payment methods from core source model are not grouped properly

### ISSUE THREAD

**Issue by wojtekn** [magento/magento2#19664]
### Summary

There is `\Magento\Payment\Model\Config\Source\Allmethods` class in the Magento core which can be used as source model for backend configuration fields. It displays available payment methods grouped by payment provider. Not all methods shipped with M2 have group defined, so some of them are displayed outside option group. This can be confusing.

It looks as follows now:

<img width="780" alt="methods-without-group-1" src="https://user-images.githubusercontent.com/727413/49687886-550d9b80-fb0a-11e8-8a14-a999cb694929.png">
<img width="777" alt="methods-without-group-2" src="https://user-images.githubusercontent.com/727413/49687888-56d75f00-fb0a-11e8-8385-fbd2fdddb0d4.png">

### Examples

1. Create empty custom module
2. Add new field to system configuration using mentioned source model:

```
    <group id="payment" translate="label" type="text" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1">
        <label>Payment settings</label>
            <field id="ccmethods" translate="label" type="multiselect" sortOrder="1" showInDefault="1" showInWebsite="1" showInStore="1">
            <label>CC methods</label>
            <source_model>Magento\Payment\Model\Config\Source\Allmethods</source_model>
        </field>
    </group>
```

### Proposed solution

Configure one missing group for PayPal method, one for Authorize.net method and a group for all Braintree methods.

I will submit PR for this soon.

**Comment by magento-engcom-team** [comment]
Hi @wojtekn. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@wojtekn do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by wojtekn** [comment]
This shows how methods look after applying the fix:

<img width="762" alt="methods-fixed" src="https://user-images.githubusercontent.com/727413/49687989-b4b87680-fb0b-11e8-9fb2-c41d48c01345.png">

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97021`, `MAGETWO-97022` were created

**Comment by magento-engcom-team** [comment]
Hi @wojtekn. Thank you for your report. 
The issue has been fixed in magento/magento2#19665 by @wojtekn in 2.3-develop branch
Related commit(s):
 - [614a7e43b58268afa572a9091fa7e6c51b7dbcc2](https://github.com/magento/magento2/commit/614a7e43b58268afa572a9091fa7e6c51b7dbcc2)
 - [9566898098dcd70d09943be5c5c81d90566c1b44](https://github.com/magento/magento2/commit/9566898098dcd70d09943be5c5c81d90566c1b44)

The fix will be available with the upcoming 2.3.1 release.
